presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/NodesResponse.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.elasticsearch;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+
+import java.util.Map;
+import java.util.Set;
+
+public class NodesResponse
+{
+    private final Map<String, Node> nodes;
+
+    @JsonCreator
+    public NodesResponse(@JsonProperty("nodes") Map<String, Node> nodes)
+    {
+        this.nodes = ImmutableMap.copyOf(nodes);
+    }
+
+    public Map<String, Node> getNodes()
+    {
+        return nodes;
+    }
+
+    public static class Node
+    {
+        private final Set<String> roles;
+        private final Http http;
+
+        @JsonCreator
+        public Node(
+                @JsonProperty("roles") Set<String> roles,
+                @JsonProperty("http") Http http)
+        {
+            this.roles = ImmutableSet.copyOf(roles);
+            this.http = http;
+        }
+
+        public Set<String> getRoles()
+        {
+            return roles;
+        }
+
+        public Http getHttp()
+        {
+            return http;
+        }
+    }
+
+    public static class Http
```

Comment:
There is some jackson magic to remove the wrapper

Reply:
I haven't being able to find anything straightforward from a cursory search. It may require writing custom deserializers.